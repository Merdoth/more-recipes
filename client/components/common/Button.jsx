import React from 'react';


const Button = ({
  type, onClick, disabled, name, iconClass, className,
}) => (
    <div
      className='form-group buttonb'
    >

      <button
        onClick={ onClick }
        type={ type }
        className={ className}
        disabled={ disabled }
        name={ name }
      >
     <i className={`fa ${iconClass}`} ></i> { name }
     </button>
    </div>
);

export default Button;
