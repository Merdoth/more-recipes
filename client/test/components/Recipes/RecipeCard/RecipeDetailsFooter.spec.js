import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import sinon from 'sinon';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedRecipeDetailsFooter,{ RecipeDetailsFooter } from
  '../../../../components/Recipes/RecipeCard/RecipeDetailsFooter.jsx';
import Button from '../../../../components/common/Button.jsx';

/* global jest */
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const setup = () => shallow(<RecipeDetailsFooter />);

describe('RecipeDetailsFooterComponent snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});


describe('RecipeDetailsFooter Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper.find(Button)).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit()', () => {
  xit('should add recipes to the state', () => {
    const event = {
      preventDefault: jest.fn()
    };
    const wrapper = setup();
    const form = wrapper.find('.cta-btn');
    wrapper.setState({
      fullName: 'ucheya'
    });

    form.simulate('submit', event);
  });
});

describe('handleDelete()', () => {
  xit('should add recipes to the state', () => {
    const event = {
      preventDefault: jest.fn()
    };
    const wrapper = setup();
    const form = wrapper.find('.cta-btn');
    wrapper.setState({
      fullName: 'ucheya'
    });

    form.simulate('submit', event);
  });
});

describe('Connected RecipeDetailsFooter component', () => {
  it('tests that the component successfully rendered', () => {
    const store = mockStore({

    });
    const wrapper = shallow(<ConnectedRecipeDetailsFooter store={store} />);
    expect(wrapper.length).toBe(1);
  });
});

