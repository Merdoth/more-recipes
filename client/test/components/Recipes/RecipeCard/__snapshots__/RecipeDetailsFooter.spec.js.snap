// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsFooter Component snapshot it should render the right amount of elements 1`] = `
<div
  className="userButton row"
>
  <Button
    className="btn btn-lg btn-primary btn-block cta-btn"
    id="edit"
    name="Edit"
    onClick={[Function]}
    type="edit"
  />
  <Button
    className="btn btn-lg btn-primary btn-block cta-btn"
    id="delete"
    name="Delete"
    onClick={[Function]}
    type="delete"
  />
</div>
`;

exports[`RecipeDetailsFooter Component snapshot it should render the right amount of elements 2`] = `
<div
  className="userButton row"
>
  <Button
    className="btn btn-lg btn-primary btn-block cta-btn"
    id="edit"
    name="Edit"
    onClick={[Function]}
    type="edit"
  />
  <Button
    className="btn btn-lg btn-primary btn-block cta-btn"
    id="delete"
    name="Delete"
    onClick={[Function]}
    type="delete"
  />
</div>
`;
